tests/components/hvv_departures/test_config_flow.py

```diff
@@ -0,0 +1,380 @@
+    CONF_REAL_TIME,
+    CONF_STATION,
+    DOMAIN,
+)
+from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL, SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_OFFSET, CONF_PASSWORD, CONF_USERNAME
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry, load_fixture
+
+FIXTURE_INIT = json.loads(load_fixture("hvv_departures/init.json"))
+FIXTURE_CHECK_NAME = json.loads(load_fixture("hvv_departures/check_name.json"))
+FIXTURE_STATION_INFORMATION = json.loads(
+    load_fixture("hvv_departures/station_information.json")
+)
+FIXTURE_CONFIG_ENTRY = json.loads(load_fixture("hvv_departures/config_entry.json"))
+FIXTURE_OPTIONS = json.loads(load_fixture("hvv_departures/options.json"))
+FIXTURE_DEPARTURE_LIST = json.loads(load_fixture("hvv_departures/departure_list.json"))
+
+
+async def test_user_flow(hass):
+    """Test that config flow works."""
+
+    with patch(
+        "homeassistant.components.hvv_departures.hub.GTI.init",
+        return_value=FIXTURE_INIT,
+    ), patch("pygti.gti.GTI.checkName", return_value=FIXTURE_CHECK_NAME,), patch(
+        "pygti.gti.GTI.stationInformation", return_value=FIXTURE_STATION_INFORMATION,
+    ), patch(
+        "homeassistant.components.hvv_departures.async_setup", return_value=True
+    ), patch(
+        "homeassistant.components.hvv_departures.async_setup_entry", return_value=True,
+    ):
+
+        # step: user
+
+        result_user = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data={
+                CONF_HOST: "api-test.geofox.de",
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+        assert result_user["step_id"] == "station"
+
+        # step: station
+        result_station = await hass.config_entries.flow.async_configure(
+            result_user["flow_id"], {CONF_STATION: "Wartenau"},
+        )
+
+        assert result_station["step_id"] == "station_select"
+
+        # step: station_select
+        result_station_select = await hass.config_entries.flow.async_configure(
+            result_user["flow_id"], {CONF_STATION: "Wartenau"},
+        )
+
+        assert result_station_select["type"] == "create_entry"
+        assert result_station_select["title"] == "Wartenau"
+        assert result_station_select["data"] == {
+            CONF_HOST: "api-test.geofox.de",
+            CONF_USERNAME: "test-username",
+            CONF_PASSWORD: "test-password",
+            CONF_STATION: {
+                "name": "Wartenau",
+                "city": "Hamburg",
+                "combinedName": "Wartenau",
+                "id": "Master:10901",
+                "type": "STATION",
+                "coordinate": {"x": 10.035515, "y": 53.56478},
+                "serviceTypes": ["bus", "u"],
+                "hasStationInformation": True,
+            },
+            "stationInformation": {
```

Comment:
Why do we store the station  information in the config entry? We should only store data that we need in the integration.

Reply:
It's not yet needed, I programmed the integration with a binary_sensor platform (for elevator status at that station) but for the smaller PR just included the sensor. But as soon as I PR the binary_sensor that information is needed and then that information is already available

Reply:
Ok, then please only save that specific information. We don't need the return code, the station outline or the last update.

Reply:
I thought about it and I can probably get around with not saving that data in the config flow data (it could theoretically change anyways...) so i removed it completely.